UN-OCHA/odsg8-site
# [ODSG-45] D10 upgrade

Refs: ODSG-45

This upgrades the ODSG site to Drupal 10 and PHP 8.2.

## Tests

**Test the update the site to D10**

1. Create/update local site to latest DB from prod
   1.  Check out the `develop` branch
   2. Optionally create a new instance with `./local/install.sh -m -c -i`)
   3. Install the dev dependencies with `./local/exec.sh -w /srv/www site composer install`
   4. Import a recent DB dump
   5. Clear the cache
   6. Import the config with `./local/exec.sh -u appuser site drush cim`
2. Check out the `feature/ODSG-45` branch
3. Recreate the with the PHP 8.2 image with `./local/install.sh -m`
4. Install the dev dependencies with `./local/exec.sh -w /srv/www site composer install`
5. Update the site to D10 with `./local/exec.sh -u appuser site drush deploy`
6. Follow the post deployment steps from https://github.com/UN-OCHA/odsg8-site/pull/227

**Test the updated site**

This should be done in parallel on the `prod` site and on the `local` site or the `dev` site.

Note: the local and dev sites have the fixed `publications` blocks from https://github.com/UN-OCHA/odsg8-site/pull/227 while that hasn't yet been deployed to prod.

1. Page access --> confirm same behavior as prod.
   1. As anonymous user, check access to several pages and files and confirm the redirection to the welcome page
   2. Do the same as an authorized users without any role
   3. Do the same as an ODSG member (access to public pages)
   4. Do the same as a content manager (access to public and private pages + access to unpublished content)
2. Appearance --> visit several pages and confirm similarity with prod with a few slight differences like the heading margins due to the update to the latest CD theme, that's normal
3. Edit some pages, make some changes and confirm they are reflected when saving the pages
